File: testers/avro/java-spring/generated-and-checked-in/com/example/events/AddressPublisher.java
package com.example.events;

import java.util.concurrent.CompletableFuture;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.kafka.support.SendResult;
import org.springframework.stereotype.Service;

/** Type-safe event publisher for address topic */
@Service
public record AddressPublisher(KafkaTemplate<String, Address> kafkaTemplate, String topic) {
  public AddressPublisher(KafkaTemplate<String, Address> kafkaTemplate) {
    this(kafkaTemplate, "address");
  }

  public AddressPublisher withKafkaTemplate(KafkaTemplate<String, Address> kafkaTemplate) {
    return new AddressPublisher(kafkaTemplate, topic);
  }

  public AddressPublisher withTopic(String topic) {
    return new AddressPublisher(kafkaTemplate, topic);
  }

  /** Publish a Address event */
  public CompletableFuture<SendResult<String, Address>> publish(String key, Address event) {
    return kafkaTemplate.send(topic, key, event);
  }
}

File: testers/avro/scala-cats/generated-and-checked-in/com/example/events/AddressPublisher.scala
package com.example.events

import cats.effect.IO
import fs2.kafka.KafkaProducer
import fs2.kafka.ProducerRecord
import fs2.kafka.ProducerRecords
import fs2.kafka.ProducerResult
import scala.annotation.unused

@unused
/** Type-safe event publisher for address topic */
case class AddressPublisher(
  kafkaTemplate: KafkaProducer[IO, String, Address],
  topic: String = "address"
) {
  /** Publish a Address event */
  def publish(
    key: String,
    event: Address
  ): IO[ProducerResult[String, Address]] = {
    return kafkaTemplate.produce(ProducerRecords.one(ProducerRecord(topic, key, event))).flatten
  }
}